jwql/website/apps/jwql/data_containers.py

```diff
@@ -887,47 +891,46 @@ def get_thumbnails_all_instruments(parameters):
     filenames = []
 
-    if parameters['instruments'] is None:
-        thumbnails = []
     for inst in parameters['instruments']:
-        print("Retrieving thumbnails for", inst)
         # Make sure instruments are of the proper format (e.g. "Nircam")
         instrument = inst[0].upper() + inst[1:].lower()
 
         # Query MAST for all rootnames for the instrument
         service = "Mast.Jwst.Filtered.{}".format(instrument)
 
-        params = {"columns": "*",
-                  "filters": [{"paramName": "apername",
-                               "values": parameters['apertures'][inst.lower()]
-                               },
-                              {"paramName": "detector",
-                               "values": parameters['detectors'][inst.lower()]
-                               },
-                              {"paramName": "filter",
-                               "values": parameters['filters'][inst.lower()]
-                               },
-                              {"paramName": "exp_type",
-                               "values": parameters['exposure_types'][inst.lower()]
-                               },
-                              {"paramName": "readpatt",
-                               "values": parameters['read_patterns'][inst.lower()]
-                               }
-                              ]}
+        if (parameters['apertures'][inst.lower()] == []) and (parameters['detectors'][inst.lower()] == []) \
+            and (parameters['filters'][inst.lower()] == []) and (parameters['exposure_types'][inst.lower()] == []) \
+            and (parameters['read_patterns'][inst.lower()] == []):
+                params = {"columns": "*",
+                          "filters": []}
+        else:
+            query_filters = []
+            if (parameters['apertures'][inst.lower()] != []):
+                if instrument != "Nircam":
+                    query_filters.append({"paramName": "pps_aper", "values": parameters['apertures'][inst.lower()]})
+                if instrument == "Nircam":
+                    query_filters.append({"paramName": "apername", "values": parameters['apertures'][inst.lower()]})
+            if (parameters['detectors'][inst.lower()] != []):
+                query_filters.append({"paramName": "detector", "values": parameters['detectors'][inst.lower()]})
+            if (parameters['filters'][inst.lower()] != []):
+                query_filters.append({"paramName": "filter", "values": parameters['filters'][inst.lower()]})
+            if (parameters['exposure_types'][inst.lower()] != []):
+                query_filters.append({"paramName": "exp_type", "values": parameters['exposure_types'][inst.lower()]})
+            if (parameters['read_patterns'][inst.lower()] != []):
+                query_filters.append({"paramName": "readpatt", "values": parameters['read_patterns'][inst.lower()]})
+            params = {"columns": "*",
+                      "filters": query_filters}
```

Comment:
Nice investigative work here to figure out how to get around the wildcard parameter!